File: Assignment-18/task18.3.cpp
#include <iostream>
using namespace std;

int factorial(int n) {
    if (n == 0)
        return 1;
    else
        return n * factorial(n - 1);
}

int main() {
    cout << "Input: 5 → Output: Factorial = " << factorial(5) << endl;
    cout << "Input: 0 → Output: Factorial = " << factorial(0) << endl;
    return 0;
}

File: Assignment-18/task18.3.py
def factorial(n):
    if n == 0:
        return 1
    else:
        return n * factorial(n - 1)

# Calling the function
print(f"Input: 5 → Output: Factorial = {factorial(5)}")
print(f"Input: 0 → Output: Factorial = {factorial(0)}")
